errcheck fails

considering running errcheck in travis-ci to prevent future failures

```sh
mischief@delta ~/code/go/src/github.com/david415/go-sphinxmixcrypto $ errcheck ./...
params.go:135:9:        h.Write(data)
```

--- a/params.go
+++ b/params.go
@@ -129,13 +129,11 @@ func (s *Params) GenerateCipherStream(key [chachaKeyLen]byte, numBytes uint) ([]
 }
 
 // HMAC authenticates our message.
-func (s *Params) HMAC(key [16]byte, data []byte) [16]byte {
+func (s *Params) HMAC(key [16]byte, data []byte) (ret [16]byte) {
 	h := blake2b.NewMAC(16, key[:])
-	h.Reset()
-	h.Write(data)
-	var ret [16]byte
+	_, _ = h.Write(data)
 	copy(ret[:], h.Sum(nil))
-	return ret
+	return
 }
 
 // EncryptBlock encrypts a block

